modules/repository-datastax/src/main/java/com/intuit/wasabi/repository/cassandra/impl/CassandraExperimentRepository.java

```diff
@@ -1253,8 +1249,13 @@ public void createApplication(Application.Name applicationName) {
 
             Map<Application.Name, Set<String>> result = new HashMap<>();
             for (ListenableFuture<Result<ExperimentTagsByApplication>> future : futures) {
-                ExperimentTagsByApplication expTagApplication = future.get().one();
-                result.put(Application.Name.valueOf(expTagApplication.getAppName()), expTagApplication.getTags());
+                List<ExperimentTagsByApplication> expTagApplication = future.get().all();
+
+                Set<String> allTagsForApplication = new TreeSet<>();
```

Comment:
I was using it mainly for the UI, such that @longdogz doesn't need to order it there, but whatever works best :) I also think the performance impact is neglectable due to the small element count.